### PR TITLE
[Ingest Manager] Prevent reporting ecs version twice

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Fixes: #20666

Due to this, we get two `ecs.versions`  in a resulting event.

e.g
```
...

        "ecs": {
            "version": "1.6.0"
        },
        "ecs.version": "1.5.0",
        "event": {
            "dataset": "elastic.agent.metricbeat"
        },
...
```

Funny thing is also that it might differ in version as in the example above. But this may be just due to different builds of agent and beat locally.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by ph** [comment]
@michalpristas I am converned.. why the version could mismatch? I believe we are using the ecs go library to generate the version is that correct? With that I would expect that beats and Agent to be on the same version?

File: x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go [hunk 1]
```diff
@@ -113,7 +113,6 @@ func (b *Monitor) EnrichArgs(process, pipelineID string, args []string, isSideca
 		logFile = fmt.Sprintf("%s-json.log", logFile)
 		appendix = append(appendix,
 			"-E", "logging.json=true",
-			"-E", "logging.ecs=true",
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: I am confused? If this is removed then what is added the ecs.version to the events?

Comment (michalpristas) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: confused as well i'm still looking, but i think this is not ideal place to fix, need to play with it a bit more

Comment (michalpristas) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: i think that this is due to monitoring beat with a beat so we monitor data which already contains ecs flags and injects additional one there using ^ settings. 
One ECS is probably comming from processing pipeline (look for `WithECS`) and additional one we were adding when logging.

This is why i see 2 versions 1.5.0 and 1.6.0. `1.6.0` is coming from publishing pipeline and version is set in libbeat.
`1.5.0` is coming from logp as it is hardcoded there. 

And as i dont want to alter publishing pipeline and libbeat code, this seems like a good place for a fix after all.

Comment (ph) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: I think it's a problem with logp, that version should be centralized. WDYT @urso

Comment (blakerouse) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: Ah, that explains why. Agent is already adding the ecs fields in the log, then filebeat is adding it again.

**Review by blakerouse — APPROVED** [review]
Approving because removing that setting is actually correct in this case. I think the ecs version issue in logo should be handled separately.

File: x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go (hunk 1 above)
Comment (ph) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: Does enabling the ecs flags do more than just adding the _version_?

Comment (michalpristas) [reply x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go:116]: i compared event with ecs enabled and without ecs and they differed in version, `host` and `agent` stayed there

**Pull request merged**